ui.go

```diff
@@ -20,35 +20,44 @@ package main
 
 import "fmt"
 
-//if it looks stupid and works, it ain't stupid
-const clearLine = "\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08"
-
 //UI encapsulates the state of the terminal display.
 type UI struct {
 	task  string
 	step  uint
 	count uint
+
+	prevLen int
+}
+
+func (ui *UI) setLine(line string) {
+	fmt.Printf("\r%-[1]*[2]s", ui.prevLen, line)
```

Comment:
What's going on here? This doesn't parse in my head, and Github's syntax highlighting flags the `%` as an error (presumably an incomplete formatting directive).

Reply:
* `-` pad with spaces on the right rather than the left (left-justify the field)
* `*` get the width to whitespace-pad to from an argument, rather than from the format string itself
* `[1]` identifies which argument to get the width from
* `[2]` identifies which argument to get the string contents from

It could be written without the `[1]` and `[2]`, but they help me when reading it.

```go
fmt.Printf("\r%-*s", ui.prevLen, line)
```